addon/components/nrg-dropdown/component.js

```diff
@@ -296,10 +296,11 @@ export default Component.extend(Validation, EKMixin, EKFirstResponderOnFocusMixi
       return false;
     }
     const isDropdownIcon = evt.target.closest('.dropdown.icon');
+    const clickedInsideDropdownElement = evt.target.closest('.dropdown');
     if (this.search && !this.isOpen) {
       this.openDropdown();
     } else if (!this.search || isDropdownIcon) {
-      if (this.isOpen) {
+      if (this.isOpen && isDropdownIcon || !clickedInsideDropdownElement ) {
```

Comment:
Can you clarify what this condition does now? I assume you want to make sure the dropdown is open, and either click on the icon or click outside the dropdown element. If so, the second two conditions should be wrapped:
```suggestion
      if (this.isOpen && (isDropdownIcon || !clickedInsideDropdownElement)) {
```

Otherwise, every click outside of the dropdown element will meet this condition since the two conditions on the `&&` will run first, before the second portion of the `||`.